src/stravalib/protocol.py

```diff
@@ -89,6 +102,168 @@ def __init__(
+                method=method, url=url, params=params
+            )
+        )
+        if params is None:
+            params = {}
+        if self.access_token:
+            params["access_token"] = self.access_token
+
+        methods = {
+            "GET": self.rsession.get,
+            "POST": functools.partial(self.rsession.post, files=files),
+            "PUT": self.rsession.put,
+            "DELETE": self.rsession.delete,
+        }
+
+        try:
+            requester = methods[method.upper()]
+        except KeyError:
+            raise ValueError(
+                "Invalid/unsupported request method specified: {}".format(
+                    method
+                )
+            )
+
+        raw = requester(url, params=params)  # type: ignore[operator]
+        # Rate limits are taken from HTTP response headers
+        # https://developers.strava.com/docs/rate-limits/
+        self.rate_limiter(raw.headers, method)
+
+        if check_for_errors:
+            self._handle_protocol_error(raw)
+
+        # 204 = No content
+        if raw.status_code in [204]:
+            resp = {}
+        else:
+            resp = raw.json()
+
+        return resp
+
+    def _token_expired(self) -> bool:
+        """Checks if a token has expired or not. Returns True if it's expired.
+
+        Returns
+        -------
+        bool
+            True if token has expired, otherwise returns false
+        """
+        if self.token_expires:
+            if time.time() > self.token_expires:
+                print("Your token has expired; Refreshing it now.")
+                return True
+            else:
+                return False
+        else:
+            logging.warning(
+                "Please make sure you're set client.token_expires if you",
+                " want automatic token refresh to work",
+            )
+            return False
+
+    def refresh_expired_token(self) -> None:
+        """Checks to see if a token has expired and auto refreshes it
+        if the user has setup their environment with the client
+        secret information.
+
+        Returns
+        -------
+        None
+            If all is setup properly, updates and resets the access_token
+            attribute. Otherwise prints a statement letting the user know that
+            they can set that up if they wish or they need to refresh their
+            token.
+
+        """
+        try:
+            # If the package is not typed, mypy is not happy, ignore
+            # https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
+            from dotenv import load_dotenv  # type: ignore
+
+            load_dotenv()
+            logging.info("Loaded .env file successfully.")
+        except ImportError:
+            logging.debug(
+                "python-dotenv is not installed. Skipping .env file loading."
+                " If you need .env support, please install `python-dotenv`."
+            )
```

Comment:
I think (in hindsight?) we should entirely leave out any possible .env file processing. Larger apps would include `stravalib` as a dependency and have their own management of development/test/production environments and would use `load_dotenv` on a higher level.

For smaller hobby projects, we could add references to our documentation on how to use .env files + pydotenv (there are plenty of great tutorials online).

The only thing our code should do is assume these env variables are set in whichever way the user prefers.

Reply:
That's a good point. I struggled with this because i didn't like adding an (optional) dependency. this will make the code much cleaner as we only need the os library to access envt variables. 

I have a doc locally where i'm building out the steps so I'll just update our tutorials accordingly and we can link to the dot file tutorials as well. It's pretty simple to setup and now I don't have to worry about tests for it!